### PR TITLE
#864 development constant tail bug extending obj

### LINKED ISSUE #864: DevelopmentConstant drops the tail factor
### Are you on the latest chainladder version?

- [x] Yes, this bug occurs on the latest version.

### Describe the bug in words

The `DevelompentConstant()` discards the last CDF to ultimate when the last CDF is not 1.0 (it sets it to 1.0).

### How can the bug be reproduced?

```python
reported_patterns = {
    12: 4.0, 24: 2.9, 36: 1.8, 48: 1.4, 60: 1.2, 72: 1.1, 84: 1.03, 96: 1.02, 108: 1.005,
}
auto_bi = cl.load_sample("friedland_auto_bi_insurer")
reported_BI_claim = cl.DevelopmentConstant(
    patterns=reported_patterns, style="cdf"
).fit_transform(auto_bi["Reported Claims"])
reported_BI_claim.cdf_
```

### What is the expected behavior?

The` 108-Ult` factor of 1.005 should not be discarded.

### PR DESCRIPTION
**Pull request by kennethshsu**
## Summary of Changes  
I started over, and now the flow should be much easier to follow.

I alos fixed the assert on `test_constant_callable_axis1`: `assert np.all(abs(lhs.values[0, :, 0, :] - patterns.values) < atol)` now compares the whole array instead of missing the last one.

## Related GitHub Issue(s)  
Fixes #864

## Additional Context for Reviewers  

| Pattern supplied compared to tri obj | Slice in obj | This PR |
| :--- | :--- | :--- |
| shorter | [:-1] | fill missing ages with 1.0 |
| one short | [:-1] | no change - old implementation |
| same length | [:] | added the tail with length 1 |
| longer | [:] | extend obj with tail columns |

- [x] I passed tests locally for both code (`uv run pytest`) and documentation changes (`uv run jb build docs --builder=custom --custom-builder=doctest`)

---

> [!NOTE]
> **Medium Risk**
> Changes core reserve development pattern fitting logic; incorrect alignment could shift LDF/CDF outputs, though behavior is heavily regression-tested.
> 
> **Overview**
> Fixes **#864** by reworking how `DevelopmentConstant.fit` aligns external CDF/LDF patterns with triangle development ages.
> 
> **`DevelopmentConstant.fit`** now branches on pattern length versus `obj.ddims`: shorter patterns keep the existing `[:-1]` slice and default missing ages to **1.0** via `patterns.get(item, 1.0)`; equal-length patterns use the full development axis; longer patterns **append tail columns** of ones, widen `ddims`, and refresh slicers so the full supplied pattern is applied. Callable patterns use the same length logic from a sample dict before building LDFs.
> 
> **Tests** add coverage for short, exact, long, tail/no-tail, incremental triangles, and CDF vs LDF; **`test_constant_callable_axis1`** now compares the full CDF array to `patterns.values` (not `[:, :-1]`).
> 
> **.gitignore** adds `.vscode/`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3bbbcd8b67569d67aa8a2d2a00926219ed0f3bbd. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by henrydingliu** [comment]
I'm looking for precedent inside the codebase and found [this](https://github.com/casact/chainladder-python/blob/31b3432589afda2a86dfea65df093affa6a047ba/chainladder/core/tests/test_triangle.py#L399). I think it's worth mimicking here. something like (in pseudo code)

```python
raa_long = raa
raa = raa_long[valuation < 1988]
raa_short = raa_long[valuation < 1986]
dev = cl.Development().fit(raa)
ldf_dict = cl.Development().fit(raa).ldf_.to_frame().to_dict
DC_ldf = cl.DevelopmentConstant(ldf_dict, 'ldf')
assert DC_ldf.fit(raa_long).ldf_ == dev.ldf_
assert DC_ldf.fit(raa_long).cdf_ == dev.cdf_
#repeated for raa and raa_short
#repeated for DC_cdf = cl.DevelopmentConstant(cdf_dict, 'cdf')
```

**Comment by kennethshsu** [comment]
@henrydingliu this is now ready for you to review

File: chainladder/development/tests/test_constant.py [hunk 1]
```diff
@@ -23,36 +23,473 @@ def test_constant_ldf(raa):
...
\ No newline at end of file
+        xerror = cl.DevelopmentConstant(
+            patterns=paid_cdfs, callable_axis=2, style="cdf"
+        ).fit(agway)
+    lhs = (
+        cl.DevelopmentConstant(patterns=paid_cdfs, callable_axis=1, style="cdf")
+        .fit(agway)
+        .cdf_
+    )
+    assert np.all(abs(lhs.values[0, :, 0, :] - patterns.values) < atol)
```
[the remark this answers is not in the document]
Comment (kennethshsu) [reply chainladder/development/tests/test_constant.py:163]: This is intentional

File: chainladder/development/constant.py [hunk 2]
```diff
@@ -53,42 +54,91 @@ def fit(self, X, y=None, sample_weight=None):
...
+        # pattern supplied is longer than the triangle
+        else:
+            obj = obj.iloc[..., 0, :] * 0 + 1
+            extra = len(pattern_ddims) - len(obj.ddims)
+            if extra > 0:
+                tail = xp.ones(obj.shape)[..., -1:]
+                tail = xp.repeat(tail, extra, -1)
+                obj.values = xp.concatenate((obj.values, tail), -1)
+                obj.ddims = np.array(pattern_ddims)
+                obj._set_slicers()
```
[the remark this answers is not in the document]
Comment (kennethshsu) [reply chainladder/development/constant.py:100]: Ehh, is this worth generalizing?

**Review by henrydingliu — APPROVED** [review]
